pyartcd/pyartcd/pipelines/operator_sdk_sync.py

```diff
@@ -0,0 +1,148 @@
+        if 'assembly' not in release_yaml['releases'][assembly]['assembly']['basis'].keys():
+            raise ValueError("Can not find jira and advisory number from assembly")
+        return self.get_ad_jira_key(release_yaml['releases'][assembly]['assembly']['basis']['assembly'], release_yaml)
+
+    def _get_sdkversion(self, build):
+        output = subprocess.getoutput(f"oc image info --filter-by-os amd64 -o json {build} | jq .digest")
+        shasum = re.findall("sha256:\\w*", output)[0]
+        cmd = f"oc image extract {constants.OPERATOR_URL}@{shasum} --path /usr/local/bin/{self.sdk}:. --confirm && chmod +x {self.sdk} && ./{self.sdk} version && rm {self.sdk}"
+        self._logger.info(cmd)
+        sdkvalue = subprocess.getoutput(cmd)
+        sdkversion = re.findall("v\\d.*-ocp", sdkvalue)[0]
+        return sdkversion
+
+    def _extract_binaries(self, arch, sdkVersion, build):
+        output = subprocess.getoutput(f"oc image info --filter-by-os {arch} -o json {build} | jq .digest")
+        shasum = re.findall("sha256:\\w*", output)[0]
+
+        rarch = arch
+        rarch = 'x86_64' if arch == 'amd64' else rarch
+        rarch = 'aarch64' if arch == 'arm64' else rarch
+        tarballFilename = f"{self.sdk}-{sdkVersion}-linux-{rarch}.tar.gz"
+
+        cmd = f"rm -rf ./{rarch} && mkdir ./{rarch}" + \
+              f" && oc image extract {constants.OPERATOR_URL}@{shasum} --path /usr/local/bin/{self.sdk}:./{rarch}/ --confirm" + \
+              f" && chmod +x ./{rarch}/{self.sdk} && tar -c --preserve-order -z -v --file ./{rarch}/{tarballFilename} ./{rarch}/{self.sdk}" + \
+              f" && ln -s {tarballFilename} ./{rarch}/{self.sdk}-linux-{rarch}.tar.gz && rm -f ./{rarch}/{self.sdk}"
+        self._logger.info(cmd)
+        subprocess.run(cmd, shell=True)
+        if arch == 'amd64':
+            tarballFilename = f"{self.sdk}-{sdkVersion}-darwin-{rarch}.tar.gz"
+            cmd = f"oc image extract {constants.OPERATOR_URL}@{shasum} --path /usr/share/{self.sdk}/mac/{self.sdk}:./{rarch}/ --confirm" + \
+                  f" && chmod +x ./{rarch}/{self.sdk} && tar -c --preserve-order -z -v --file ./{rarch}/{tarballFilename} ./{rarch}/{self.sdk}" + \
+                  f" && ln -s {tarballFilename} ./{rarch}/{self.sdk}-darwin-{rarch}.tar.gz && rm -f ./{rarch}/{self.sdk}"
+            self._logger.info(cmd)
+            subprocess.run(cmd, shell=True)
+        self._sync_mirror(rarch)
+
+    def _sync_mirror(self, arch):
+        extra_args = "--exclude '*' --include '*.tar.gz'"
+        local_dir = f"./{arch}/"
+        if self.prerelease:
+            s3_path = f"/pub/openshift-v4/{arch}/clients/operator-sdk/pre-release/"
+        else:
+            s3_path = f"/pub/openshift-v4/{arch}/clients/operator-sdk/{self.assembly}/"
+        cmd = f"aws s3 sync --no-progress --exact-timestamps {extra_args} --delete {local_dir} s3://art-srv-enterprise{s3_path}"
+        self._logger.info(cmd)
+        subprocess.run(cmd, shell=True)
```

Comment:
The exit code is not captured here.

Reply:
we don't need to capture exit code, if any command fails in cmd it should fail the job and we need to check the print logs.

Reply:
> if any command fails in cmd it should fail the job

That is not true. There are 2 issues with using shell script in `subprocess.run`:
- Unless `check=True` is set, `subprocess.run` won't raise an exception on error.
- If multiple commands are executed in a shell script, some shell options like `set -eo pipefail` should be set otherwise the shell script won't abort or return non-zero.